proxy/proxy.go

```diff
@@ -57,56 +58,94 @@ type BaseURLResolver interface {
 // 	- logging errors and proxy request timing to stdout
 //
 // Note that this will panic if `resolver` is nil.
-func NewHandlerFunc(timeout time.Duration, resolver BaseURLResolver) http.HandlerFunc {
+func NewHandlerFunc(config types.FaaSConfig, resolver BaseURLResolver) http.HandlerFunc {
 	if resolver == nil {
 		panic("NewHandlerFunc: empty proxy handler resolver, cannot be nil")
 	}
 
-	proxyClient := http.Client{
+	proxyClient := NewProxyClientFromConfig(config)
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Body != nil {
+			defer r.Body.Close()
+		}
+
+		switch r.Method {
+		case http.MethodPost,
+			http.MethodPut,
+			http.MethodPatch,
+			http.MethodDelete,
+			http.MethodGet:
+
+			proxyRequest(w, r, proxyClient, resolver)
+
+		default:
+			w.WriteHeader(http.StatusMethodNotAllowed)
+		}
+	}
+}
+
+// NewProxyClientFromConfig creates a new http.Client designed for proxying requests and enforcing
+// certain minimum configuration values.
+func NewProxyClientFromConfig(config types.FaaSConfig) *http.Client {
+	maxIdleConns := config.MaxIdleConns
+	if maxIdleConns < 1 {
```

Comment:
Same for this

I'd rather see this encapsulated in the parsing code or in a method like:

With either A)
```
config.GetMaxIdleConns()
```

or B)

The value is defaulted within the code that loads/parses the `types.FaaSConfig`